starter_code/elevator.js

```diff
@@ -1,19 +1,96 @@
 class Elevator {
-  constructor(){
-    this.floor      = 0;
-    this.MAXFLOOR   = 10;
-    this.requests   = [];
+  constructor() {
+    this.floor = 0;
+    this.MAXFLOOR = 10;
+    this.requests = [];
+    this.direction = "";
+    this.waitingList = [];
+    this.passengers = [];
   }
 
-  start() { }
-  stop() { }
-  update() { }
-  _passengersEnter() { }
-  _passengersLeave() { }
-  floorUp() { }
-  floorDown() { }
-  call() { }
-  log() { }
+  start() {
+    this.startId = setInterval(() => {
+      this.update();
+    }, 1000);
+  }
+  stop() {
+    clearInterval(this.startId);
+  }
+  update() {
+    // if (this.passengers.length === 0 && this.requests.lenth !== 0) {
+    //   if (this.floor < this.requests[this.requests.length - 1]) {
+    //     this.direction = "up";
+    //     while (this.floor < this.requests[this.requests.length - 1]) {
+    //       this.floorUp();
+    //     }
+    //   } else if (this.floor > this.requests[this.requests.length - 1]) {
+    //     while (this.floor > this.requests[this.requests.length - 1]) {
+    //       this.floorDown();
+    //     }
+    //   }
+    // }
+    for (let i = 0; i < this.MAXFLOOR; i++) {
```

Comment:
The way I would look at `update`, is as a function that is called at every interval, and does the logic based on the *current state* of the elevator instance. So, instead of iterating over all the floors at every interval, I would just:
1. Check who needs to enter at current floor and let them enter
2. Check who needs to exit at current floor and let the exit
3. Check if we need to keep going in the same direction or change
4. Go up one floor or down one floor, depending on the direction after step 3.
5. That's it, this will be performed again at the next floor.